proactive-jupyter-kernel/kernel.py

```diff
@@ -1962,6 +2021,53 @@ def __submit_job__(self, input_data):
         else:
             input_data['name'] = self.job_name
 
+        if self.added_nodesource_submitjob:
+            self.proactive_job.removeGenericInformation("NODESOURCENAME")
```

Comment:
Should we not test if the GI `NODESOURCENAME` is not yet set?

Reply:
We already do it. The variable self.added_nodesource_submitjob is set to True only if the nodesource is added using submitJob pragma.